src/components/ui/DateTimeInput/DateTimeInput.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { Text, TouchableOpacity, View } from "react-native";
+import DateTimePickerModal from "react-native-modal-datetime-picker";
+import { Icon } from "@app/components/ui";
+import { useStyles } from "./styles";
+import dayjs from "dayjs";
+
+interface DateTimeInputProps {
+  id: string;
+  value?: string; // GMT format
+  onConfirm: (date: string) => void;
+  errorMessage?: string;
+  placeholder?: string;
+}
+
+const DateTimeInput: React.FC<DateTimeInputProps> = ({
+  id,
+  value,
+  onConfirm,
+  errorMessage,
+  placeholder,
+}) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [currentDate, setCurrentDate] = useState<Date | undefined>(undefined);
+  const styles = useStyles({ date: currentDate });
+
+  useEffect(() => {
+    if (value) {
+      setCurrentDate(new Date(value));
+    }
+  }, [value]);
+
+  const showDatePicker = useCallback(() => {
+    setIsVisible(true);
+  }, []);
+
+  const hideDatePicker = useCallback(() => {
+    setIsVisible(false);
+  }, []);
+
+  const handleConfirm = useCallback(
+    (date: Date) => {
+      onConfirm(date.toUTCString());
+      setCurrentDate(date);
+      hideDatePicker();
+    },
+    [hideDatePicker, onConfirm],
+  );
+
+  const displayText = useMemo(() => {
+    if (currentDate) {
+      return dayjs(currentDate).format("DD MMMM [at] hh:mm a");
```

Comment:
not sure how dayjs works but does this format invalid string formats?

Reply:
Good catch. it doesn't check for invalid dates atm. I just assume whatever is passed is a valid date string